### PR TITLE
halloffame credits

### LINKED ISSUE #41: hall of fame image broken
https://geokrety.org/halloffame.php
bottom of this page includes broken image for tools.
img_20180412_210109

---
work branch: [feature/halloffame](https://github.com/geokrety/geokrety-website/tree/feature/halloffame)

---
- [x] improve config 'icon_width'
- [x] integrate into target page ([demo](https://boly38.gk.kumy.org/halloffame.php))
- [x] move icons files to CDN and update related config url
- [x] squash, rebase, PR

### PR DESCRIPTION
**Pull request by boly38**
- add way to generate credits table + unit test
- common test function in TestUtil
- add config-credits
- improve config icon_size, icon_style
- integrate into target page halloffame
- use CDN for logos files geokrety/GeoKrety-Static#2
- add CONFIG_CDN_LOGOS
- fixes #41

### CODE REVIEW

**Comment by boly38** [comment]
demo > [boly38](https://boly38.gk.kumy.org/halloffame.php)

seems we are ok
(only diff with boly38 branch is commit wording (miss '#' to link issue)

File: website/lib/geokrety/view/Credits.class.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        }
+
+        return count($this->credits) - 1;
+    }
+
+    public function toHtmlDivs() {
+        $creditsDivs =
+        '<div class="dcreds">';
+        for ($i = 1; $i < count($this->credits); ++$i) {
+            unset($creditDesc,$creditLink,$iconUrl);
```
[the remark this answers is not in the document]
Comment (boly38) [reply website/lib/geokrety/view/Credits.class.php]: I think no becasue i passed phpcs, maybe a rule to add somewhere

**Comment by boly38** [comment]
- [ ] add @Lineflyer to translators heros :)

File: tests/CreditsTest.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+class CreditsTest extends TestCase {
+    public function test_generate_credits() {
+        // GIVEN
+        define('CONFIG_CDN_LOGOS', 'https://cdn.geokrety.org/images/logos/');
```
[the remark this answers is not in the document]
Comment (boly38) [reply tests/CreditsTest.php]: (for test context) would like to isolate as much as possible target component, but that generates one duplicate we could maybe accept

**Comment by boly38** [comment]
- [ ] space others

**Comment by boly38** [comment]
I don't have any idea on how to fix that :

image

**Comment by Lineflyer** [comment]
Thanks for adding me. :)
I suggest to take a list of top x contributors of the geokrety crowdin project. I am IMHO not the only contributor.

**Comment by kumy** [comment]
See https://crowdin.com/project/geokrety/settings#reports-top-members for the list of contributors.

**Comment by kumy** [comment]
> I don't have any idea on how to fix that :

@boly38 Seems like an edge case not handled by the PMD rule set. Maybe adding an [exclude](https://phpmd.org/documentation/suppress-warnings.html) is your best bet, also I didn't found an issue for this case on their [bug tracker](https://github.com/phpmd/phpmd/issues)

**Comment by Lineflyer** [comment]
@kumy 403 for me on this link. I guess its only for the project owner

**Comment by boly38** [comment]
image

I disable PMD for this file because PMD doc is not clear: where is the rule defined, what is his name ? 
escaping test echo is useless.. 

I think I will merge in this state

**Pull request merged**